### ISSUE TITLE
incorrect license should be caught

### ISSUE THREAD

**Issue by h1alexbel** [objectionary/lints#354]
In XMIR, we have `<license/>` node with a program's license. We should compare this license with one supplied in `LICENSE.txt` in the root of the repo. If it does not match, we should report it with `WARNING` severity.

**Comment by h1alexbel** [comment]
@yegor256 @maxonfjvipon WDYT?

**Comment by yegor256** [comment]
@h1alexbel maybe it's better to make our XMIR more compliant with [SPDX](https://en.wikipedia.org/wiki/Software_Package_Data_Exchange), and instead of `license` have this:

```
<program>
  <license name="MIT">2023-2025 Yegor Bugayenko</license>
</program>
```

**Comment by h1alexbel** [comment]
@yegor256 good point, we need to implement this on both layers: EO and lints. @maxonfjvipon FYI

**Comment by yegor256** [comment]
@h1alexbel what `spdx-is-not-top-meta` is about?

**Comment by h1alexbel** [comment]
@yegor256 its about position of `+spdx` meta among other metas. It should be at the highest level (1-2 line numbers).

This is incorrect:

```eo
+architect jeff@google.com
+spdx SPDX-License-Identifier MIT

# F.
[] > f
```

While this is a correct one:

```eo
+spdx SPDX-License-Identifier MIT
+architect jeff@google.com

# F.
[] > f

```

In other words, `+spdx` meta should be out of default meta sort order.

**Comment by yegor256** [comment]
@h1alexbel what's the point? why can't it be among other metas? we need metas to be sorted.

**Comment by h1alexbel** [comment]
@yegor256 I feel like `+spdx` meta should be at the top of the file (like license header that it substitutes), or its redundant?
